lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,33 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds qw(ONE_HOUR);
+
+# Remove or rename these "constants" in favor of a dynamic path
+use constant MAX_DURATION_SECONDS => ONE_HOUR / 2;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app) {
+    path($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache') . '/gitlab_maintenance.lock');
```

Comment:
Instead of a hardcoded `/var/lib/openqa` I would use
```
use OpenQA::Utils qw(prjdir);
```
Also, is this `cache` a new directory? It doesn't exist on o3, for example.
As a result, `$file->touch` below will fail because it needs to have that directory first.

Looking in our source code, `/var/lib/openqa/cache` is the default directory for the cache service (which runs on workers).

It might be confusing to use the same directory here.